src/test/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoTest.java

```diff
@@ -396,6 +404,45 @@ public void deleteMessage() throws DatabaseAccessException, DocumentException {
                     verify(batch).delete(same(documentReference));
                 }
 
+                @Test
+                public void reportMessage() throws DatabaseAccessException, DocumentException, InvalidOperationException {
+                    mockGetGroupAndForumIds();
+                    given(
+                        collectionAdapter.getDocumentsWhereEqualTo(anyString(), anyString(), any(), anyString(), any()))
+                        .willReturn(query);
+                    given(documentSnapshot.getReference()).willReturn(documentReference);
+                    given(documentSnapshot.get(anyString())).willReturn(usernameList);
+                    given(batch.update(any(DocumentReference.class), anyString(), any())).willReturn(batch);
+
+                    dao.reportMessage(groupName, forumName, username, username2, date);
+                    verify(collectionAdapter)
+                        .getDocumentsWhereEqualTo(eq(Path.ofCollection(Collections.messages, groupId, forumId)),
+                            eq("creator"), eq(username), eq(publicationDate), eq(date));
+                    verify(batch).update(same(documentReference), eq("reportedBy"),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "reportedBy" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5633182)